hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/StaleRecoveringContainerScrubbingService.java

```diff
@@ -84,8 +85,11 @@ static class RecoveringContainerScrubbingTask implements BackgroundTask {
 
     @Override
     public BackgroundTaskResult call() throws Exception {
-      containerSet.getContainer(containerID).markContainerUnhealthy();
-      LOG.info("Stale recovering container {} marked UNHEALTHY", containerID);
+      Container con = containerSet.getContainer(containerID);
+      if (null != con) {
+        con.markContainerUnhealthy();
```

Comment:
For recoveringFeature, it seems parallel container is removed causing it to be null. Inside this marking unhealthy, it will write to file which might have been deleted.
IMO, we should check inside also if container really exist after lock before update unhealthy.